src/regressionmodel.jl

```diff
@@ -93,3 +93,17 @@ function predict! end
 Return the residual degrees of freedom of the model.
 """
 function dof_residual end
+
+"""
+    reconstruct(model::RegressionModel, newY)
+
+Reconstruct explanatory variables from `model` given new response values `newY`.
```

Comment:
For consistency with `predict`, shouldn't `newY` be optional, at least in theory (implementations can do what they want)?

```suggestion
    reconstruct(model::RegressionModel[, newY])

Reconstruct explanatory variables from `model`.
An object with new response values `newX` can be supplied, which should have
the same type and structure as the output of [`predict(model)`](@ref).
```

Reply:
It's fine to have this parameter optional. In general, it would mean to return original data. 🤷